utils/prof_utils.py

```diff
@@ -0,0 +1,1177 @@
+def find_minima_maxima(profile, ignore_threshold=0, min_comp_len=0):
+    """
+    Finds all minima and maxima of the input profile. Assumes that the profile has noise zero-clipped.
+
+    profile: list
+        The profile with noise zero-clipped
+    ignore_threshold: float
+        OPTIONAL -  Maxima with values below this number will be ignored. Default: 0
+    min_comp_len: float
+        OPTIONAL - Minimum length of a component to be considered real. Measured in bins. Default: 0
+
+    Returns:
+    --------
+    minima: list
+        A list of floats corresponding to the bin location of the profile minima
+    maxima: list
+        A list of floats corresponding to the bin location of the profile maxima
+    """
+    #If there is more than one component, find each one
+    comp_dict, comp_idx = find_components(profile, min_comp_len)
+
+    maxima=[]
+    minima=[]
+    #loop over each profile component
+    for key in comp_dict.keys():
+        x = np.linspace(0, len(comp_dict[key])-1, len(comp_dict[key]), dtype=int)
+        spline = UnivariateSpline(x, comp_dict[key], s=0.0, k=4)
+        comp_roots = spline.derivative().roots()
+        # These are the roots, we want to split maxima and minima ^^
+        comp_maxima=[]
+        comp_minima=[]
+        for i, root in enumerate(comp_roots):
+            idx = int(root)
+            left = comp_dict[key][idx-1]
+            if left>comp_dict[key][idx]:
+                comp_minima.append(root)
+            else:
+                comp_maxima.append(root)
+        #Turn the root locations into locations on profile, not on component
+        for root in comp_minima:
+            abs_root = root + comp_idx[key][0]
+            minima.append(abs_root)
+        for root in comp_maxima:
+            abs_root = root + comp_idx[key][0]
+            maxima.append(abs_root)
+
+    ignore_idx = []
+    for i, mx in enumerate(maxima):
+        if max(profile[int(mx-1):int(mx+1)]) < ignore_threshold*max(profile):
+            ignore_idx.append(i)
+    for i in sorted(ignore_idx, reverse=True):
+        del maxima[i]
+
+    return minima, maxima
+
+#---------------------------------------------------------------
+def find_maxima_err(model, noise_std, maxima):
+    """
+    Finds the uncertainty in a maxima by using the noise_std to find the distance to the points on the model that intersect at maxima-noise_std
+
+    Parameters:
+    -----------
+    model: list
+        The model to test
+    noise_std: float
+        The standard deviation of the noise from the profile used to make the model
+    maxima: list
+        The location of the maximum points in bin numbers
+
+    Returns:
+    --------
+    maxima_err: list
+        A list with length equal to that of the maxima list containing an uncertainty value for each maximum point
+    """
+    x = np.linspace(0, len(model)-1, len(model))
+    maxima_err = []
+    for i, mx in enumerate(maxima):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused variable 'i'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)